src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -1,30 +1,37 @@
 package org.broadinstitute.dsde.workbench.leonardo.model
 
+import java.net.URL
 import java.time.Instant
 import java.util.UUID
 
 import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes
 import com.typesafe.config.ConfigFactory
 import net.ceedubs.ficus.Ficus._
 import org.broadinstitute.dsde.workbench.leonardo.config.{DataprocConfig, ProxyConfig}
 import org.broadinstitute.dsde.workbench.leonardo.model.ClusterStatus.ClusterStatus
-import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.{GoogleBucket, GoogleBucketUri, GoogleProject, GoogleServiceAccount}
-import spray.json.{DefaultJsonProtocol, DeserializationException, JsString, JsValue, JsonFormat}
+import org.broadinstitute.dsde.workbench.leonardo.model.GoogleBucketUri.scheme
+import org.broadinstitute.dsde.workbench.leonardo.model.StringValueClass.LabelMap
+import spray.json.{DefaultJsonProtocol, DeserializationException, JsString, JsValue, JsonFormat, SerializationException}
 
 import scala.language.implicitConversions
 
-// maybe we want to get fancy later
-object ModelTypes {
-  type GoogleProject = String
-  type GoogleServiceAccount = String
-  type GoogleBucket = String
-  type GoogleBucketUri = String
-}
-
-object GoogleBucketUri {
-  def apply (bucketName: String, fileName: String): GoogleBucketUri = {
-    s"gs://${bucketName}/${fileName}"
-  }
+// this needs to be a Universal Trait to enable mixin with Value Classes
+// it only serves as a marker for StringValueClassFormat
+sealed trait StringValueClass extends Any
```

Comment:
It might be good to add `with Product with Serializable` to `StringValueClass`.

What it does is make case class type inference work better. It's actually sometimes needed to make more complicated stuff compile; at the very least, it fixes an annoyance. Observe:
```
sealed trait StringValueClass extends Any
case class ClusterName(string: String) extends AnyVal with StringValueClass
case class GoogleProject(string: String) extends AnyVal with StringValueClass
// what is the inferred type of x?
val x = if (scala.util.Random.nextBoolean) ClusterName("foo") else GoogleProject("bar")
// ugh
x: Product with Serializable with AnyVal with StringValueClass = GoogleProject(bar)
```
Compare with:
```
sealed trait StringValueClass extends Any with Product with Serializable
case class ClusterName(string: String) extends AnyVal with StringValueClass
case class GoogleProject(string: String) extends AnyVal with StringValueClass
// what is the inferred type of x?
val x = if (scala.util.Random.nextBoolean) ClusterName("foo") else GoogleProject("bar")
// better
x: StringValueClass with AnyVal = GoogleProject(bar)
```

Reply:
Also, nothing in the trait requires that it be a String value class. (It could be an Int value class). Maybe name it `LeoValueClass` or something?

Reply:
I am not sure I'm seeing the benefit of adding Product/Serializable here.

I do want this trait to be specifically for Strings, to enable generic JSON processing (and Mockito matching).  I didn't see a way to enforce it while still conforming to the Universal Trait requirements.